Confusing readme.

The readme says to do:

```
import { HttpProvider } from "../bxsolana/provider/http.js"
```

but there's nothing in `../bxsolana` ... that'd be like, if I was working inside a `git clone` of the repo, not if I'm working inside my own project that did a `npm i` of the module, right?

This is confusing, what's the correct way of importing this module?

If I do instead:

```
import { HttpProvider } from '@bloxroute/solana-trader-client-ts';
```

then I get:

```
Module '"@bloxroute/solana-trader-client-ts"' has no exported member 'HttpProvider'.ts(2305)
```

Thanks.

EDIT:

This works (at least produces no typescript errors, have not tried running it yet)

```
import { HttpProvider } from "node_modules/@bloxroute/solana-trader-client-ts/bxsolana/provider/http";
```

but it's a pretty dirty hack. What's the "correct" way of doing this?

EDIT2:

Nope, not working, I get:

```
src/bloxroute-pumpfun-buy-sell.ts:4:38 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("node_modules/@bloxroute/solana-trader-client-ts/bxsolana/provider/http")' call instead.
  To convert this file to an ECMAScript module, change its file extension to '.mts', or add the field `"type": "module"` to '/home/arthur/dev/ai/dragunov/package.json'.

4 import { HttpProvider         } from "node_modules/@bloxroute/solana-trader-client-ts/bxsolana/provider/http";
```

Also tried dynamic imports, they also fail:

```
    // Dynamically import the bloxroute modules.
    // Using 'any' type to bypass TypeScript module resolution issues
    const httpModule: any = await import("node_modules/@bloxroute/solana-trader-client-ts/bxsolana/provider/http");
    const constantsModule: any = await import("node_modules/@bloxroute/solana-trader-client-ts/bxsolana/utils/constants");
    
    // Extract the imported items.
    const HttpProvider = httpModule.HttpProvider;
    const MAINNET_API_NY_HTTP = constantsModule.MAINNET_API_NY_HTTP;
```

--- a/README.md
+++ b/README.md
@@ -21,16 +21,161 @@ In `node.js` environments, you can specify both of these values in a `.env` file
 the browser, you'll want to define them during run-time from user input, and probably use some wallet provider to handle
 the transaction signing.
 
-A simple example:
+## Examples
+
+<!-- GrpcClient Example -->
+<details open>
+<summary>GrpcClient</summary>
 
 ```typescript
-import { HttpProvider } from "../bxsolana/provider/http.js"
+import {
+    MAINNET_API_GRPC_PORT,
+    MAINNET_API_NY_GRPC,
+    GrpcProvider,
+    GetRecentBlockHashRequest,
+    loadFromEnv
+} from "@bloxroute/solana-trader-client-ts";
 
-const provider = new HttpProvider()
+// Calls to provider must be made inside async function
+async function main(): Promise<void> {
+    try {
+        // Load configuration from environment variables
+        const config = loadFromEnv();
 
-let req = await provider.getOrderbook({ market: "SOLUSDC", limit: 5 })
-console.info(req)
+        // Initialize the GrpcProvider with necessary credentials and endpoint
+        const provider = new GrpcProvider(
+            config.authHeader,
+            config.privateKey,
+            `${MAINNET_API_NY_GRPC}:${MAINNET_API_GRPC_PORT}`,
+            true
+        );
+
+        // Prepare the request for fetching the recent block hash
+        const request: GetRecentBlockHashRequest = {};
+
+        // Fetch the recent block hash from the provider
+        const response = await provider.getRecentBlockHash(request);
+
+        // Log the response
+        console.info("Recent Block Hash Response: ");
+        console.info(JSON.stringify(response, null, 2));
+    } catch (error) {
+        console.error("Error fetching recent block hash:", error);
+    }
+}
+
+// Execute the main function
+main();
 ```
+</details>
+
+---
+
+<!-- HttpClient Example -->
+<details>
+<summary>HttpClient</summary>
+
+```typescript
+import {
+  MAINNET_API_NY_HTTP,
+  HttpProvider,
+  GetRecentBlockHashRequest,
+  loadFromEnv
+} from "@bloxroute/solana-trader-client-ts";
+import { AxiosRequestConfig } from "axios"
+
+// Calls to provider must be made inside async function
+async function main(): Promise<void> {
+  try {
+    // Load configuration from environment variables
+    const config = loadFromEnv();
+    
+    const requestConfig: AxiosRequestConfig = {
+      timeout: 30_000,
+    }
+    
+    // Initialize the HttpProvider with necessary credentials and endpoint
+    let provider = new HttpProvider(
+      config.authHeader,
+      config.privateKey,
+      MAINNET_API_NY_HTTP,
+      requestConfig
+    )
+    
+    // Prepare the request for fetching the recent block hash
+    const request: GetRecentBlockHashRequest = {};
+    
+    // Fetch the recent block hash from the provider
+    const response = await provider.getRecentBlockHash(request);
+    
+    // Log the response
+    console.info("Recent Block Hash Response: ");
+    console.info(JSON.stringify(response, null, 2));
+  } catch (error) {
+    console.error("Error fetching recent block hash:", error);
+  }
+}
+
+// Execute the main function
+main();
+```
+
+</details>
+
+---
+
+<!-- WsClient Example Example -->
+<details>
+<summary>WsClient</summary>
+
+```typescript
+import {
+  MAINNET_API_NY_WS,
+  WsProvider,
+  GetRecentBlockHashRequest,
+  loadFromEnv
+} from "@bloxroute/solana-trader-client-ts";
+
+// Calls to provider must be made inside async function
+async function main(): Promise<void> {
+  try {
+    // Load configuration from environment variables
+    const config = loadFromEnv();
+    
+    // Initialize the WsProvider with necessary credentials and endpoint
+    const provider = new WsProvider(
+      config.authHeader,
+      config.privateKey,
+      MAINNET_API_NY_WS
+    )
+    
+    // Prepare the request for fetching the recent block hash
+    const request: GetRecentBlockHashRequest = {};
+    
+    // Connect to the WebSocket server
+    await provider.connect()
+    
+    // Fetch the recent block hash from the provider
+    const response = await provider.getRecentBlockHash(request);
+    
+    // Close the connection
+    provider.close()
+    
+    // Log the response
+    console.info("Recent Block Hash Response: ");
+    console.info(JSON.stringify(response, null, 2));
+  } catch (error) {
+    console.error("Error fetching recent block hash:", error);
+  }
+}
+
+// Execute the main function
+main();
+```
+
+</details>
+
+---
 
 Refer to the `examples/` for more info. As mentioned above, you'll need an `.env` file for exported variables to execute
 the full suite. A proper `.env` file looks like something like this.

